src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Auth.Actions.ManageUsersActions.ManageUsersPatchSingle;
+using Core.Database;
+using Core.Database.Enums;
+using Core.Database.Models.Auth;
+using Core.Exceptions;
+using Core.Services;
+using HashidsNet;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Auth.Actions.ManageUsersActions.ManagerUsersPatchSingle
+{
+    public class ManageUsersPatchSingleService
+    {
+        private readonly AppDbContext _appDbContext;
+        private readonly RoleService _roleService;
+        private readonly Hashids _hashids;
+        private readonly ILogger<ManageUsersPatchSingleService> _logger;
+
+        public ManageUsersPatchSingleService(
+            AppDbContext appDbContext,
+            RoleService roleService,
+            Hashids hashids,
+            ILogger<ManageUsersPatchSingleService> logger
+            )
+        {
+            _appDbContext = appDbContext;
+            _roleService = roleService;
+            _hashids = hashids;
+            _logger = logger;
+        }
+
+        public async Task<ManageUsersPatchSingleResponseItem> PatchUserRolesAndWarehouseId(string id, ManageUsersPatchSinglePayload payload)
+        {
+            await _roleService.ThrowIfNoRole(RoleName.Administrator);
+            var userId = _hashids.DecodeSingleLong(id);
+
+            var dbResult = await _appDbContext.Users
+                 .Include(c => c.Roles)!
+                     .ThenInclude(c => c.Role)
+                 .Include(c => c.Warehouse)
+                 .SingleOrDefaultAsync(c => c.Id == userId);
+
+            if (dbResult == null)
+            {
+                _logger.LogError("Couldn't find user by id in database");
+                throw new ItemNotFoundException("Sorry we are unable to find this user in database");
+            }
+
+            if (!payload.Roles.Any())
+            {
+                _logger.LogError("There were no roles in the payload");
+                throw new ClientInputErrorException("Sorry we couldn't fetch roles");
+            }
```

Comment:
No czemu, przecież rozmawialiśmy o tym. 
Jeśli ktoś nie będzie miał ról w bazce, to przy próbie naprawy poprzez PATCH, kod będzie zawsze tu wpadać i nie będziemy w stanie tego naprawić.

Reply:
tzn. zmieniłem z bazy na payload, żeby w momencie gdy chcesz wykonac patcha, ale w payloadzie nie ma ról żadnych, to wtedy ma bład wyrzucić, ale w sumie to nie będzie można Patch'em usunąć ról. Tylko nadać conajmniej 1.

Reply:
Sorry mój blad